### PR TITLE
[Model] Add support for openai/privacy-filter

### PR DESCRIPTION
**Pull request by fjosw**
## Purpose

Adds support for [`openai/privacy-filter`](https://huggingface.co/openai/privacy-filter) (`OpenAIPrivacyFilterForTokenClassification`): a gpt-oss-style MoE encoder (GQA 14/2, YaRN RoPE, 128 experts top-4, attention sinks) repurposed as a bidirectional token classifier for PII detection. Every layer uses non-causal attention with a banded ±`sliding_window` mask. The LM head is replaced with a 33-class BIOES score head.

The new model reuses gpt-oss by making the implementation more flexible (sliding window policy, NeoX / GPT-J RoPE style). `GptOssForCausalLM` behavior is unchanged.

Two preliminary fixes are bundled in:

- `flash_attn.py`: `_forward_encoder_attention()` omitted `s_aux=self.sinks` while the decoder path passes it, so encoder layers with sinks silently dropped them.
- `fused_moe/oracle/unquantized.py`: `flashinfer_cutlass_moe` aliases `SWIGLUOAI` to plain `Swiglu` on the unquantized branch. Demote both FlashInfer backends when `activation==SWIGLUOAI` so Triton's `swigluoai_and_mul` is chosen (same pattern as the Qwen3.5/`dp_size` demotion just above).

AI-assisted, every line reviewed and tested by the submitter. `gh search` for `OpenAIPrivacyFilter`, `openai_privacy_filter`, `privacy-filter`, and `banded attention` returned no duplicates.

## Test Plan

```bash
.venv/bin/python -m pytest tests/models/language/pooling/test_token_classification.py -k privacy -v
```

Seven prompts covering all eight PII categories, compared against `AutoModelForTokenClassification`.

## Test Result

H100, bf16:

- 7/7 prompts within `atol=0.1, rtol=1e-2` (max softmax diff: 0.099).
- 91/91 token argmax match vs HF.

### CODE REVIEW

**Comment by noooop** [comment]
Thanks for your contribution!  Great to have this!

**Comment by fjosw** [comment]
Just a quiet bump in case this slipped off the queue. @noooop @DarkLight1337 @WoosukKwon @bbrowning, would you (or anyone else with bandwidth) have a moment to take a look? No urgency on my side.

_There are a few open Gemini comments about pipelining which don't apply to this small classifier model imo._

File: vllm/model_executor/models/gpt_oss.py [hunk 1]
```diff
@@ -132,9 +136,25 @@ def __init__(
...
+        config: GptOssConfig,
+        cache_config: CacheConfig | None,
+        quant_config: QuantizationConfig | None,
+        prefix: str,
+    ) -> Attention:
+        # Override to swap in an encoder-only attention or alter the
+        # per-layer sliding-window policy.
         # Only apply sliding window to every other layer
         sliding_window = config.sliding_window if self.layer_idx % 2 == 0 else None
-        self.attn = Attention(
```
Comment (Isotr0py) [inline vllm/model_executor/models/gpt_oss.py:138]: I think we can deal with `EncoderOnlyAttention` like llama etc:
https://github.com/vllm-project/vllm/blob/9b343dd4f54a9870f3ba1e41f5a5b3f4a1e25340/vllm/model_executor/models/llama.py#L205-L221

File: vllm/model_executor/models/openai_privacy_filter.py [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+            positions=positions,
+            intermediate_tensors=intermediate_tensors,
+            inputs_embeds=inputs_embeds,
+        )
+        hidden_states = hidden_states.to(self.head_dtype)
+        return self.score(hidden_states)
+
+    def load_weights(self, weights: Iterable[tuple[str, torch.Tensor]]) -> set[str]:
+        loader = AutoWeightsLoader(self)
+        return loader.load_weights(weights, mapper=self.hf_to_vllm_mapper)
```
Comment (Isotr0py) [inline vllm/model_executor/models/openai_privacy_filter.py]: I think you can preprocess w13 weights and bias before `loader.load_weights` calling

**Comment by fjosw** [comment]
Thanks for the review @Isotr0py . I addressed the w13 weights and bias pre-processing comment and reconfirmed that results are identical to the huggingface transformers reference.

Regarding encoder attention: the benefit of _build_attention is that it keeps all three privacy-filter-specific variations (encoder-only attention, GPT-J RoPE, and the different sliding-window rule) together in one place. Your suggestion (as I understand it) would mean plumbing attn_type through three classes, and we'd still need separate hooks for the RoPE convention and the sliding-window rule. I see a benefit in keeping the variations centralized via the single override, but happy to switch to the llama-style plumbing if you prefer.

_The pre-commit test has been repeatedly failing on `Update Dockerfile dependency graph`, I think this is not related to any of my changes._

**Comment by Isotr0py** [comment]
> Your suggestion (as I understand it) would mean plumbing attn_type through three classes

No, we just need to consider encoder-only and decoder-only for GPT-OSS now. GPT-J RoPE is not coupled with the `EncoderOnlyAttention` and `Attention`, while sliding-window will be handled automatically inside the attention interface. So there is still one override attention class for GPT-J RoPE:

```python3
class OpenAIPrivacyFilterAttention(OAIAttention):
    # Privacy-filter uses GPT-J style RoPE (interleaved pairs), not NeoX.
    rope_is_neox_style = False
```

I meant we can directly remove the extra `_build_attention` method if we choose llama-style plumbing.

File: vllm/model_executor/models/gpt_oss.py [hunk 3]
```diff
@@ -221,6 +241,10 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
 
 
 class TransformerBlock(torch.nn.Module):
+    # Override to swap attention/MLP without re-implementing the block.
+    attention_cls: type[nn.Module] = OAIAttention
+    mlp_cls: type[nn.Module] = MLPBlock
```
Comment (Isotr0py) [inline vllm/model_executor/models/gpt_oss.py:247]: I think there's no need to expose `mlp_cls` since they're the same.

**Review by Isotr0py — APPROVED** [review]
Anyway, current status LGTM as well.

**Review by noooop — APPROVED** [review]

**Comment by noooop** [comment]
Ci is currently using transformers==5.5.3, but openai/privacy-filter requires 5.6.0. I will try to merge this PR when transformers is upgraded to 5.6.0.

cc @hmellor

**Review by mgoin — APPROVED** [review]

**Comment by hmellor** [comment]
FYI vLLM should be updating to Transformers 5.10.4 soon in CI
